packages/amplify_datastore_plugin_interface/test/testData/Blog.dart

```diff
@@ -13,19 +13,28 @@
 * permissions and limitations under the License.
 */
 
+// ignore_for_file: public_member_api_docs
+
+import 'ModelProvider.dart';
 import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
 import 'package:collection/collection.dart';
 import 'package:flutter/foundation.dart';
 
-import 'ModelProvider.dart';
-
 /** This is an auto generated class representing the Blog type in your schema. */
 @immutable
 class Blog extends Model {
   static const classType = const BlogType();
   final String id;
-  final String name;
-  final List<Post>? posts;
+  final String? _name;
+  final List<Post>? _posts;
+
+  String get name {
+    return _name!;
```

Comment:
We are already discussing this one offline, but I don't see how this could be a safe cast.

Reply:
Yup it's not.  The Blog graphqL schema has a required name field.  Customers wouldn't expect a required field to be nullable.  However, we make the underlying private field nullable to handle the fact that the model might not have that field if it was a nested model with only the id field set.

Reply:
I think it is fine for now since datastore won't be in the initial preview and since this looks like it is just for tests. I assume there will be more discussion about datastore offline.